app/src/main/java/com/android/brewr/ui/overview/Overview.kt

```diff
@@ -82,13 +82,15 @@ fun OverviewScreen(
   }
 
   LaunchedEffect(permissionGranted) {
-    if (permissionGranted) {
+    if (permissionGranted && !isFetched) {
       coroutineScope.launch {
-        val currentLocation =
-            withContext(Dispatchers.IO) { getCurrentLocation(context) } ?: LatLng(46.5197, 6.6323)
-        fetchNearbyCoffeeShops(
-            coroutineScope, context, currentLocation, onSuccess = { coffeeShops = it })
+        getCurrentLocation(
+            context,
+            onSuccess = {
+              fetchNearbyCoffeeShops(coroutineScope, context, it, onSuccess = { coffeeShops = it })
+            })
```

Comment:
Great! This one looks more efficient than fetching the same coffee shops multiple times